### ISSUE TITLE
BUG when skipping previous classes.

### ISSUE THREAD

**Issue by Iniciando** [HerculesWS/Hercules#1603]
By using a script from the initial class to the final class of the game and delivering skill points to the player via the "SkillPoint" script command, the receiving player can ignore the order of point distribution from the initial classes to the final classes .

**Example**:
I am a Novice (Job_Novice_High -ID: 4001) and I transform myself via script to the third final class Warlock (Job_Warlock_T - ID: 4061), so I will gain 186 skill points, 59 of "jobchange 4061" + 127 "SkillPoint ":
9 - Novice skill points (Job_Novice_High -ID: 4001) "Basic skills".
49 - Mage skill points (Job_Mage_High - ID: 4003) - 1rt
69 - High Wizard Skill Points (Job_High_Wizard - ID: 4010) - 2nd
59 - Warlock Skill Points (Job_Warlock_T - ID: 4061) - 3rd
Total = 186 points.
These points need to be distributed in the order of lowest to highest, and there are certain amount to be distributed in the home class, as shown above. However, when this "jump" occurs, the emulator fails to make these restrictions and the player can distribute those skill points in any way they want.

**Example**:
I am a Warlock (Job_Warlock_T - ID: 4061) and I have 186 points to distribute and as the emulator can not understand this "jump" I will distribute this way.
9 - Novice skill points (Job_Novice_High -ID: 4001) "Basic skills".
8 - Mage skill points (Job_Mage_High - ID: 4003) - 1rt
70 - High Wizard Skill Points (Job_High_Wizard - ID: 4010) - 2nd
99 - Warlock Skill Points (Job_Warlock_T - ID: 4061) - 3rd.

**Example script**:

```
Prontera, 150,175.0 example script 521, {
If (Class == 4001) {
End;
}
Switch (select ("Warlock.T", "Rune Knight.T", "Ranger .T")) {
Case 1:
Jobchange 4061;
Skillpoint + 127;
End;
Case 2:
Jobchange 4060;
Skillpoint + 127;
End;
Case 3:
Jobchange 4062;
Skillpoint + 127;
End;
}
}
```

**Comment by Emistry** [comment]
in official server  you have no way to skip jobchange, it's simply your custom script issues.

**Comment by Iniciando** [comment]
The script does not influence how the points will be distributed, it only uses the jobchange command, which will define this is the source.

**Comment by iRagno** [comment]
When change the job there are two variables that automatically sets:

jobchange_level : from first to second job
jobchange_level_3rd : from second to third job

Those vars saves the job level that player has when changed job and are used to calculate the number of skill points player has to assign in 1st and 2nd tab, needed when used Neuralizer.

There is not issue at all, if you change job from novice job level 1 to warlock, jobchange_level saves that value and then the requeriment is to assign (var-1) skill points to first job skills, 0 in case player changes at job level 1.

What you can do is to manually sets values for those vars, but have on mind that may have weird results, since it's something abnormal. You may need to disconnect player after changing the job, to update change vars values.

**Comment by Iniciando** [comment]
Good, I understood what you mean However, we need to get to the ideal point, because I'm not the only one having access to this jobchange command, but all the server owners, and they have the same problems. And the tricky thing is that it's not something used on official servers (except for GMs). What I see many doing is after taking the players to the final class through jobchange, is to give @allskills, so it is not necessary for the player to distribute any skill points because they have all the skills. But I personally do not think that's right. I see it as a BUG, but the decision is up to you.

**Comment by iRagno** [comment]
There is no way emulator can know how to deal with that job change, in the way that you talked to bring 139 SkillPoints in an arbitrary way. Why 139? why not 129? why not 88?

Emulator does what should do, and saves the job level when job changes. What you ask is an unnatural behaviour and can be done, but as Emistry said it should be fixed in script, not in emulator.

I propose you this example:

```
prontera,150,175,0	script	example script	521,{
	if (Class != 4001)
		end;
	switch (select("Warlock.T", "Rune Knight.T", "Ranger .T")) {
	case 1:
		jobchange 4061;
		break;
	case 2:
		Jobchange 4060;
		break;
	case 3:
		Jobchange 4062;
		break;
	}
	SkillPoint + 137;
	jobchange_level = 50;
	jobchange_level_3rd = 70;
	end;
}
```

In this way you could do what you need, but it need test and may need to kick player, since those vars are not updated in same instant.

**Comment by Iniciando** [comment]
Me, I just finished the test and it really worked, but as you said, I had to replay the player. But I do not particularly see any problem with that. I already had this problem for a long time, and always when opening threads in the forum ngm knew what happened, so I had already given up. Now I see how resolved and apologize for the topic and for having insisted it was a BUG. Thank you for your attention and help me find the real source of the problem. Greetings!
